src/Compilers/Core/Portable/MetadataReference/MetadataReference.cs

```diff
@@ -257,9 +257,15 @@ public MetadataReference WithProperties(MetadataReferenceProperties properties)
         /// Reusing <see cref="AssemblyMetadata"/> object allows for sharing data accross these references.
         /// </para>
         /// </remarks>
+        [Obsolete("Use CreateFromFile(assembly.Location) instead", error: true)]
         public static MetadataReference CreateFromAssembly(Assembly assembly)
```

Comment:
Please put `[EditorBrowsable(EditorBrowsableState.Never)]` on these too, so that they don't show up in IntelliSense.